### PR TITLE
Filter commandline

### LINKED ISSUE #140: Use command line input window instead of current input window
Current filter window implementation is ugly and unstable.

https://github.com/Shougo/ddu-ui-ff/issues/139

### PR DESCRIPTION
**Pull request by Shougo**
Fix #140 

Current problem:

* ~~If you execute `openFilterWindow` action in autocmd, the redraw does not work why.~~

### CODE REVIEW

File: autoload/ddu/ui/ff.vim [hunk 1]
```diff
@@ -457,21 +432,64 @@ function ddu#ui#ff#_stop_debounce_timer(timer_name) abort
...
+
+  const input = exists('*cmdline#input') ?
+        \ cmdline#input(a:params.prompt, a:input) :
+        \ input(a:params.prompt, a:input)
+
+  augroup ddu-ff-filter
+    autocmd!
+  augroup END
+  let s:filter_prev_input = input
+  doautocmd User Ddu:ui:ff:closeFilterWindow
```
Comment (kuuote) [inline autoload/ddu/ui/ff.vim:456]: Display warning when using Vim, because `Ddu:ui:ff:closeFilterWindow` fires after `ddu-ff-filter` group is disposed.

**Review by kuuote — CHANGES_REQUESTED** [review]

File: autoload/ddu/ui/ff.vim (hunk 1 above)
Comment (Shougo) [reply autoload/ddu/ui/ff.vim]: Fixed.

File: denops/@ddu-uis/ff.ts [hunk 2]
```diff
@@ -959,7 +883,7 @@ export class Ui extends BaseUi<Params> {
       await this.#saveCursor(args.denops, bufnr, cursorPos[1]);
 
       // Change real cursor
-      await args.denops.call("ddu#ui#ff#_cursor", cursorPos[1], 0);
+      await fn.call(args.denops, cursorPos[1], 0);
```
Comment (kuuote) [inline denops/@ddu-uis/ff.ts:886]: Occurs `Error: Failed to call call(2, 0): Vim:E714: List required (code: 0)`.
Maybe `fn.cursor`?

File: denops/@ddu-uis/ff.ts [hunk 3]
```diff
@@ -1001,7 +925,7 @@ export class Ui extends BaseUi<Params> {
...
```
Comment (kuuote) [inline denops/@ddu-uis/ff.ts:928]: ditto.

**Review by kuuote — CHANGES_REQUESTED** [review]

**Comment by Shougo** [comment]
Please test the branch.

**Comment by Shougo** [comment]
I will merge it in the weekend.

**Comment by Shougo** [comment]
Merged.

**Pull request merged**